src/main/java/ca/corefacility/bioinformatics/irida/service/impl/IridaFileStorageFactoryImpl.java

```diff
@@ -0,0 +1,60 @@
+package ca.corefacility.bioinformatics.irida.service.impl;
+
+import java.nio.file.Path;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.CloudSequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.LocalSequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.repositories.filesystem.IridaFileStorageService;
+
+/**
+ * Service implementation for creating SequenceFiles {@link LocalSequenceFile} and {@link CloudSequenceFile}
+ */
+@Service
+public class IridaFileStorageFactoryImpl {
```

Comment:
You can check to see but this might not be necessary if you could do all the stuff from this file in the `IridaFileStorageService` instead.  You could maybe use that as the factory.  Play with it to see though.

Reply:
Removed the factory and moved the code to the iridafilestorageservice implementations in
c8672d656aba541db4571a30762c2c670de2ab59